tools/backup_analysis.py

```diff
@@ -30,6 +30,76 @@
 
 import fnmatch, glob, os, sys, time
 
+total_blocks_global = 0
+total_bytes_global = 0
+granularity = 0
+pct20 = granularity // 5
+pct80 = pct20 * 4
+
+# There should be 1:1 matching between the names and types.
+global_names = ['Replicated Collections', 'Replicated Indexes', 'OpLog', 'Local tables', 'System tables']
+global_types = ['coll', 'index', 'oplog', 'local', 'system']
+
+assert(len(global_names) == len(global_types))
+
+blocks_global = dict()
+bytes_global = dict()
+files_changed = dict()
+files_global = dict()
+gran_blocks_global = dict()
+pct20_global = dict()
+pct80_global = dict()
```

Comment:
I think what may help is a class, like:
```
class TypeStats(object):
   def __init__(self, name):
      self.name = name
      self.blocks = 0
      self.bytes = 0
      self.files = 0
      ...etc...
```
To get it initialized,
```
typestats = dict()
typestats['coll'] = TypeStats('Replicated Collections')
typestats['index'] = TypeStats('Replicated Indexes')
...
```
Later, you can do:
```
      ts = typestats[count_type]
      ....
      ts.bytes += partial_cmp
      ....
      ts.pct20 += 1

Reply:
Thank you! That is so much cleaner.